source/frontend/src/components/common/mapFSM/machineDefinition/mapMachine.ts

```diff
@@ -372,20 +343,50 @@ const sideBarStates = {
 };
 
 const advancedFilterSideBarStates = {
-  initial: 'fullScreen',
+  initial: 'close',
   states: {
-    fullScreen: {
+    close: {
+      on: {
+        TOGGLE_FILTER: {
+          target: 'filtering',
+        },
+        TOGGLE_LAYERS: {
+          target: 'layerControl',
+        },
+      },
+    },
+    layerControl: {
       on: {
-        OPEN_ADVANCED_FILTER_SIDEBAR: {
-          target: 'sidebarOpen',
+        TOGGLE_FILTER: {
+          target: 'filtering',
+        },
+        TOGGLE_LAYERS: {
+          target: 'close',
+        },
+        SET_MAP_LAYERS: {
+          actions: assign({ activeLayers: (_, event: any) => event.activeLayers }),
         },
       },
     },
-    sidebarOpen: {
-      entry: assign({ mapFilter: () => defaultPropertyFilter }),
+    filtering: {
+      entry: [send({ type: 'REFRESH_PROPERTIES', searchCriteria: defaultPropertyFilter })],
+      exit: [send({ type: 'REFRESH_PROPERTIES' })],
       on: {
-        CLOSE_ADVANCED_FILTER_SIDEBAR: {
-          target: 'fullScreen',
+        TOGGLE_FILTER: {
+          target: 'close',
+          actions: [assign({ showDisposed: () => false }), assign({ showRetired: () => false })],
+        },
+        TOGGLE_LAYERS: {
+          target: 'layerControl',
+        },
+        SET_VISIBLE_PROPERTIES: {
+          actions: assign({ activePimsPropertyIds: (_, event: any) => event.propertyIds }),
+        },
+        SET_SHOW_DISPOSED: {
```

Comment:
Having this as an action still feels fishy to me - given that there are multiple other property states that have selectors, but we only have explicit actions for these two.

Reply:
I agree, however, this logic was in place before the refactor. I rather leave it like that and figure out later a better approach